exercises/concept/top-scorers/.meta/design.md

```diff
@@ -0,0 +1,29 @@
+# Design
```

Comment:
Would you like to add something for the analyzer?
It's a bit tricky because there are often several ways to solve a problem, but for the ones we are sure there is a best practice, we could enforce that (For example, making sure `filter` is used in `removeInsignificantPlayers` or `merge` in `combineGames`)

Reply:
Good idea.

How about this for a list of analyser checks?

- `filter` must be used in `removeInsignificantPlayers`
- `insert` must be used in `resetPlayerCounts`
- `sort` must not be used in `formatPlayers`
- `updateGoalCountForPlayer` and `List.foldl` must be used in `aggregateScorers`
- `Maybe.map` and `Maybe.withDefault` should be used in `formatPlayer`
- `merge` must be used in `combineGames`

Reply:
Yeah those are all good ones and they are in the hints (except for not using `sort` in `formatPlayers`, maybe we could add a hint for that).

Reply:
Cool, I'll create an issue on the analyser repo for these, and add a hint for `sort`.